.projenrc.js

```diff
@@ -157,6 +157,79 @@ const project = new web.ReactTypeScriptProject({
       ],
     },
   });
+
+  // Set up a canary that tests that the latest code on our main branch
+  // works against data on https://constructs.dev.
+  //
+  // We need to use the version of the front-end from the main branch
+  // and not the version that is live, otherwise it's possible for newer
+  // tests to run against an older (incompatible) version of the frontend
+  // due to deployment delays and give us false positives.
+  const e2eCanary = project.github.addWorkflow("e2e-canary");
+  e2eCanary.on({
+    schedule: [
+      {
+        cron: "0 */3 * * *", // run every three hours
+      },
+    ],
+    workflowDispatch: {},
+  });
+  e2eCanary.addJobs({
+    test: {
+      name: "constructs.dev canary",
+      runsOn: "ubuntu-latest",
+      permissions: {
+        checks: "write",
+        contents: "read",
+      },
+      steps: [
+        {
+          name: "Checkout",
+          uses: "actions/checkout@v2",
+        },
+        {
+          name: "Cypress Run",
+          uses: "cypress-io/github-action@v2",
+          with: {
+            start: "yarn proxy-server:ci",
+            "wait-on": "http://localhost:3000",
+            "wait-on-timeout": 150,
+          },
+        },
+        {
+          name: "Upload screenshots on failure",
+          uses: "actions/upload-artifact@v2",
+          if: "failure()",
+          with: {
+            name: "cypress-screenshots",
+            path: "cypress/screenshots",
+          },
+        },
+        {
+          name: "Upload videos",
+          uses: "actions/upload-artifact@v2",
+          if: "always()",
+          with: {
+            name: "cypress-videos",
+            path: "cypress/videos",
+          },
+        },
```

Comment:
It seems like these `steps` are the same that we use on the cypress PR workflow. I think it would make sense to consolidate the two into a single `cypressSteps` array